.github/workflows/github-actions.yml

```diff
@@ -4,30 +4,66 @@ on:
   push:
     paths-ignore:
       - 'README.md'
-      - 'Dockerfile'
-      - '.github/**'
       - '.dockerignore'
       - 'docker-compose.yml'
       - 'ansible/**'
   pull_request:
-    paths-ignore:
-      - 'README.md'
-      - 'Dockerfile'
-      - '.github/**'
-      - '.dockerignore'
-      - 'docker-compose.yml'
-      - 'ansible/**'
+    branches:
+      -  main
 
-jobs: 
+jobs:
   build:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-      name: Checkout the code
 
-    - run: docker build --target ci-test --tag todo-app:ci-test .
-      name: Build the CI test image
+    - name: Build the CI test image
+      run: docker build --target ci-test --tag todo-app:ci-test .
+
+    - name: Run the CI test image
+      run: docker run --env-file ./.env_test todo-app:ci-test
+      
+  publish:
+    name: Push image to Docker Hub
+    needs: build
+    runs-on: ubuntu-latest
+    if: ${{ (github.event_name == 'push') && (github.ref == 'refs/heads/main') }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      
+      - name: Build the CI prod image 
+        run: docker build --target production --tag simeonp901/todo_app:latest --tag simeonp901/todo_app:"$GITHUB_SHA" .
+
+      - name: Push the CI prod image to Docker Hub
+        run: docker push simeonp901/todo_app --all-tags
+
+  deploy:
+    name: Deploy image to Heroku      
+    needs: publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: akhileshns/heroku-deploy@v3.12.12
+        with:
+          heroku_api_key: ${{secrets.HEROKU_API_KEY}}
+          heroku_app_name: "todo-app-simeon"
+          heroku_email: "simeon_penev@hotmail.com"
+          usedocker: true
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+
+      - name: Build the CI prod image for Heroku
+        run: docker build --target production --tag registry.heroku.com/todo-app-simeon/web .
+
+      - name: Push the image to Heroku's Doker Hub registry
+        run: docker push registry.heroku.com/todo-app-simeon/web
 
-    - run: docker run --env-file ./.env_test todo-app:ci-test
-      name: Run the CI test image
\ No newline at end of file
+      - name: Release the image to Heroku
+        run: heroku container:release web -a todo-app-simeon
```

Comment:
This reads like you're running the same steps twice; once using the heroku-deploy action and then another time manually. Is that intentional?

Reply:
Thanks. Resolved.